### PR TITLE
feat(BILL-CPC-1060): integrated filter customer bills by status onto customer main page

### PR DESCRIPTION
**Pull request by MercuryAtom31**
… page

**JIRA:** link to jira ticket

https://champlainsaintlambert.atlassian.net/browse/CPC-1060

## Context:
What is the ticket about and why are we doing this change.
## Does this PR change the .vscode folder in petclinic-frontend?:

It does not.

If the PR changes the .vscode folder, explain why in detail because it should not. 
Be sure to include cgerard321 as a reviewer.

<span style="color:red">**Reviewers need to check for any changes to the 
.vscode folder and add a comment about it to their review comments.**</span>

## Changes
What are the various changes and what other modules do those changes affect.
This can be bullet point or sentence format.

I have modified the billing main page frontend, by modifying the BillsListTable.tsx file. I have added the filtering customer bills by status onto the main billing page, instead of having it on a separate page, as it was previously implemented.

## Before and After UI (Required for UI-impacting PRs)
If this is a change to the UI, include before and after screenshots to show the differences.
If this is a new UI feature, include screenshots to show reviewers what it looks like. 

BEFORE:
Screenshot 2024-09-26 at 22 18 29
AFTER:

Screenshot 2024-09-28 at 11 16 29
Screenshot 2024-09-28 at 11 16 41
Screenshot 2024-09-28 at 11 16 47
09e89a4">

## Dev notes (Optional)
Specific technical changes that should be noted
## Linked pull requests (Optional)
Pull request links

### CODE REVIEW

**Comment by ctsaki25** [comment]
I believe something may have went wrong with your git commands. It should be merging your branch into main, and not main into your branch

**Comment by Valthefirst** [comment]
> I believe something may have went wrong with your git commands. It should be merging your branch into main, and not main into your branch

That happens when you update your branch

**Comment by ctsaki25** [comment]
Ah yes, that may be it. My mistake.

**Comment by Valthefirst** [comment]
No worries

File: package-lock.json
Comment (Futurespast) [inline package-lock.json:1]: What's the changes to this file here? Did you update a dependency?

Comment (MercuryAtom31) [reply package-lock.json:1]: I did not change this file. The only thing I did was consulting the file to ascertain that Prettier was listed under devDependencies before configuring IntelliJ to recognize the installed Prettier package.

Comment (Futurespast) [reply package-lock.json:1]: Hmmm, I think it's fine. I'm just scared it might break something.

Comment (Futurespast) [reply package-lock.json:1]: @cgerard321 will this break the project?

**Comment by cgerard321** [comment]
> > I believe something may have went wrong with your git commands. It should be merging your branch into main, and not main into your branch
> 
> That happens when you update your branch

It only says merged if you did a git pull which does a merge instead of a rebase.

File: package-lock.json
Comment (MercuryAtom31) [reply package-lock.json:1]: Indeed, I use IntelliJ. I am going to open it on VS code.

Comment (MercuryAtom31) [reply package-lock.json:1]: @cgerard321 Miss, I thought ESLint was just for TypeScript files or JavaScript. However, the file in question (package-lock.json) is a JSON file. Would I need instead to format it using Prettier?

**Comment by cgerard321** [comment]
It's for everything in the front end.

File: package-lock.json
Comment (cgerard321) [reply package-lock.json:1]: It should have formatted automatically. It's possible that someone opened it in Intellj and Intellij reformatted it and they pushed it that way. 
Currently json files are not ignored by prettier.

**Review by Futurespast — DISMISSED** [review]
if it's just reformatting the it should be fine

**Review by ctsaki25 — DISMISSED** [review]
Your frontend looks good, im glad you made it show all bills by default. I don't see your backend here but im assuming it's the same changes you made in your other PR. Make sure you resolve the package-lock.json conflict (ignore yours and keep the original).

**Review by Futurespast — APPROVED** [review]
Nice, you fixed the conflicts!

**Pull request closed without merging**